Source/automap.h

```diff
@@ -14,10 +14,21 @@
 
 namespace devilution {
 
+enum MapExplorationType : uint8_t {
+	/** unexplored map tile */
+	MAP_EXP_NONE,
+	/** map explored by a shrine */
+	MAP_EXP_SHRINE,
+	/** map tile explored by someone else in multiplayer */
+	MAP_EXP_OTHERS,
+	/** map tile explored by current player */
+	MAP_EXP_SELF,
+};

```

Comment:
I think we should rethink this a bit.

First, I'd rename it to something like `AutomapTileExplorationState`.

Then I think we should consider different names for the entries, and having it as a flags enum:
- `Uncharted` for the default value. This makes sense to me considering we are talking about a map
- `Charted` to indicate if the tile is known to the player (the shrine effect would open up tiles in this state)
- `Explored` to mean a tile that has been visited

If later on we wanna keep track of "who explored", I think this should be converted into a discriminated union using `variant<T>` and adding a field to the `Explored` state to point to the player who found it.

Reply:
I see no reason to separate it like this 🤷 using current system it's easy to add new exploration types and colors